### ISSUE TITLE
Broken links after upgrade to 3.0.0

### ISSUE THREAD

**Issue by LKay** [rackt/react-router-redux#260]
After upgrade `react-router-redux` from `2.1.0` to `3.0.0` linking using `react-router` components (`<Link>`, `<IndexLink>`) seem to be broken. Components from `react-router-bootstrap` library to wrap these components (`<LinkContainer>` and `<IndexLinkContainer>`) also seem broken. 

In version `2.1.0` everything works fine and when clicking on elements the route transition is triggered. After upgrading to `3.0.0` none of the links trigger route transition but just redirects browser using `href` attribute present in generated html elements.

My versions of other libraries are:
* `react-router@2.0.0-rc5`
* `react-router-bootstrap@0.20.1`
* `redux@3.2.1`
* `react-redux@4.2.1`

**Comment by timdorr** [comment]
Nothing significant changed with the internals. Did you upgrade any other libraries at the same time? An example repo would be helpful. 

@taion Have you seen anything like this?

**Comment by taion** [comment]
I haven't, no.

**Comment by LKay** [comment]
After struggling for a while i reverted all changes in library versions and started updating libraries one by one. The above configuration causing error for `3.0.0` but when I upgraded some other libraries: `history` from `2.0.0-rc3` to `2.0.0` and `redux` from `4.2.1` to `4.3.0` then `react-router-redux@3.0.0` started working fine. 

I tried debugging this but still can't figure out what is causing the problem with links. Maybe some cross dependencies?
